dask/blockwise.py

```diff
@@ -1343,6 +1344,7 @@ def rewrite_blockwise(inputs):
     new_axes = inputs[root].new_axes
     concatenate = inputs[root].concatenate
     dsk = dict(inputs[root].dsk)
+    seen = defaultdict(int)
```

Comment:
Do we need to track this globally? Or could we start fresh at each layer we process (is this the `for i, (dep, ind) in enumerate(indices)` loop?)?

I had thought the problem was specifically when one layer used the same input name multiple times, but with different iteration orders (indices), not just that the same name occurred in the whole graph multiple times. So maybe this is too wide of a net? Or maybe it doesn't matter?

Reply:
If this is the case, then maybe rather than a counter, we could actually just have `use_dep = f"{dep}_{ind}"`, i.e. identify dependencies internally not just by unique name, but by unique combo of name and indexing pattern?

Reply:
I was wondering the same thing. Currently sitting in a REPL trying to see if I can make a name/index pair that *should* be combined with this, but is not.

Reply:
I agree that we can probably move all the logic into the same dep-processing step. This exact suggestion is one of the reasons the PR was still marked as draft. We may also want to avoid "duplicating" the k-v pair if the name+index combiunation matches exactly.

Reply:
Actually, scratch my last comment... A single `for i, (dep, ind) in enumerate(indices):` loop does not actually cover the repeated `dep` case that we need. This outer `while changed:` loop actually constructs the fused subgraph in a depth-first way (not in a breadth-first layer-by-layer order). This is why `dep` may be the same for two distinct loops, and we want to make sure we do not use the first index used for that name for all subsequent occurrences.